sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnectionCache.java

```diff
@@ -265,7 +261,31 @@ Retry retryWhenSpec(AmqpRetryPolicy retryPolicy) {
         }));
     }
 
-    private static AmqpShutdownSignal createShutdownSignal(String message) {
-        return new AmqpShutdownSignal(false, false, message);
+    /**
+     * Often, any connection closure (due to network error) happens outside of cache, and the cache needs to only check
+     * if cached connection is in closed state to see if it needs to be refreshed. But there are two cases when cache
+     * needs to close a connection explicitly,
+     * <ol>
+     * <li>When the cache itself is disposed then any cached connection or connection being cached needs to be closed
+     * explicitly. Such a cache disposal happens when all clients sharing the connection are closed.</li>
+     * <li>If all downstream cancels subscription to the cache while waiting for a new connection to connect and active,
+     * then that connection needs to be closed explicitly.</li>
+     * </ol>
+     *
+     * @param logger the logger to log the closing of connection.
+     * @param message the message to log.
+     * @param c the connection to close.
+     */
+    private static void closeConnection(ClientLogger logger, String message, ReactorConnection c) {
+        withConnectionId(logger, c).log("closing connection (" + message + ").");
+        // Listen and log any terminal signals here, though when connection close finishes, closeAsync()::Mono only
+        // emits "completion terminal signal" no matter if there was an error while closing or not.
+        c.closeAsync().subscribe(__ -> {
+        }, t -> logger.info("connection close finished with error.", t),
+            () -> logger.info("connection close finished."));
```

Comment:
maybe

```suggestion
            () -> logger.verbose("connection close finished."));
```

?

Reply:
Thanks Liudmila!. It was an intentional use of _info_. Cache disposal is not a frequent event, much lower than new-connection events.  We want to see if there are rare cases where Processor stops and if disposal path contributing to it.